### DISCUSSION TITLE
Date picker month

### DISCUSSION THREAD

**Discussion by Eralmidia** [IgniteUI/igniteui-angular#10451] [Q&A]
## Question  
After upgrading to igx 12, there seems to be quite a few changes to the date picker. We format our dates in a speficic format using a formatter like so:
![image](https://user-images.githubusercontent.com/12936286/137135454-32aacd5e-dbc8-4b41-881f-608379aa4e14.png)
This still seems to be working fine. However, when I set focus to the input field, the date is changed to a numeric value. 
![image](https://user-images.githubusercontent.com/12936286/137135932-9c8f497a-28c5-439c-a86b-5f4b8200907d.png)

I can see there is an inputFormat input, but what should I set this to in order to keep text based months?

![image](https://user-images.githubusercontent.com/12936286/137135800-48a4204b-5ae3-4d4c-acbc-e5c9b04ca485.png)

 * igniteui-angular version: 12

**Comment by jackofdiamond5** [comment]
Hi @Eralmidia, the `IgxDatePickerComponent` supports custom `inputFormat` and `displayFormat`. The former is intended to be used only while editing and it is restricted to showing numeric dates. The latter is to be used when the component is not focused (and no custom formatting is applied) and it supports all formats outlined in Angular's [DatePipe](https://angular.io/api/common/DatePipe#custom-format-options).

**Comment by Eralmidia** [comment]
@jackofdiamond5 Thank you. So as of version 12, there is now no way to actually show Oct instead of 10 while editing?

**Comment by jackofdiamond5** [comment]
@Eralmidia, yes - while editing, words in the month section have been dropped in favor of numeric values since the picker now uses the [IgxDateTimeEditorDirective](https://www.infragistics.com/products/ignite-ui-angular/angular/components/date-time-editor#display-and-input-format) internally.

**Comment by Eralmidia** [comment]
That's really unfortunate. My product manager will almost certainly dislike this change. 

It's not possible to use just the "drop down part" of the date picker and implement your own input instead either?

**Comment by Eralmidia** [comment]
Here's another issue with the new design: We have a grid with a column of dates displayed like this:
![image](https://user-images.githubusercontent.com/12936286/137438135-40f386d7-4a33-4804-ab9d-e62e5c33f58a.png)

In the previous version of the date picker, where we had templates, we listened to the paste event on the input field:
![image](https://user-images.githubusercontent.com/12936286/137438446-38801e35-5b9c-4985-b1c6-5e2ccef39e82.png)
In this paste event we took the time part of the date string and inserted into the accompanying time picker (when a date was pasted into the date input). 
![image](https://user-images.githubusercontent.com/12936286/137439154-b4670c1b-6fd9-4bd8-912f-b580fa8e000f.png)


How do you achieve something like this now? The current implementation seems to paste the date OK, but obviously the time picker isn't notified of a paste in the date input.
